runtime/caml/frame_descriptors.h

```diff
@@ -23,35 +23,75 @@
 
 #include "config.h"
 
-#define Hash_retaddr(addr, mask)                          \
-  (((uintnat)(addr) >> 3) & (mask))
-
-/* Structure of frame descriptors */
+/* The compiler generates a "frame descriptor" for every potential
+ * return address. Each loaded module has a block of memory consisting
+ * of these frame descriptors concatenated. Each frame descriptor
+ * includes:
+ *
+ * - the stack frame size, in bytes;
+ *
+ * - whether it is the return address of a call into the garbage collector,
+ *   and if so the sizes of all objects to be allocated by this call;
+ *
+ * - whether we have debug information for this stack frame, and if so
+ *   the "debuginfo" (source location) of this return address. (For
+ *   GC return addresses, this is an array of debuginfo, one for each
+ *   of the set of allocations performed by this GC entry).
+ *
+ * - the register or stack frame offset of every "live" value,
+ *   which should be scanned by the garbage collector if a GC is
+ *   performed at this point.
+ */
 
 typedef struct {
   uintnat retaddr;
-  unsigned short frame_size;
-  unsigned short num_live;
-  unsigned short live_ofs[1 /* num_live */];
+  uint16_t frame_size;
```

Comment:
Should we maybe rename the `frame_size` field if it does not in fact contain only the frame-size, and should not be used outside the `frame_descriptors` module anymore? Maybe `frame_data` or `frame_header` would work better.